src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Directives.cs

```diff
@@ -242,11 +247,16 @@ protected void BaseTypeDirective(string noTypeNameError, Func<string, SpanCodeGe
             {
                 Accept(remainingWs);
             }
+            var remainingWhitespaceLength = Span.GetContent().Value.Length;
+
             AcceptWhile(IsSpacingToken(includeNewLines: false, includeComments: true));
 
             if (EndOfFile || At(CSharpSymbolType.WhiteSpace) || At(CSharpSymbolType.NewLine))
             {
-                Context.OnError(CurrentLocation, noTypeNameError);
+                Context.OnError(
+                    CurrentLocation,
```

Comment:
Yup, you'll need to grab the starting location of the keyword. You can get at that value at the top of this block with `Span.Start` (I think).